core/src/avm2/bytearray.rs

```diff
@@ -0,0 +1,340 @@
+use flate2::read::*;
+use flate2::Compression;
+use gc_arena::Collect;
+use std::cmp;
+use std::convert::{TryFrom, TryInto};
+use std::io;
+use std::io::prelude::*;
+use std::io::{Error, ErrorKind};
+
+#[derive(Clone, Collect, Debug)]
+#[collect(no_drop)]
+pub enum Endian {
+    Big,
+    Little,
+}
+
+#[derive(Clone, Collect, Debug)]
+#[collect(no_drop)]
+pub struct ByteArrayStorage {
+    /// Underlying ByteArray
+    bytes: Vec<u8>,
+
+    // The current position to read/write from
+    position: usize,
+
+    /// This represents what endian to use while reading data.
+    endian: Endian,
+
+    /// TODO: Make this do something
+    shareable: bool,
```

Comment:
Is there a particular reason why this field is here? I can't see where it's in use and the comment implies that it has no purpose yet.